src/sentry/tasks/post_process.py

```diff
@@ -405,20 +405,34 @@ def post_process_group(
             # performance issue
             return
 
-        update_event_group(event)
         _capture_event_stats(event)
 
-        bind_organization_context(event.project.organization)
+        group_events: Mapping[int, GroupEvent] = {
+            ge.group_id: ge for ge in list(event.build_group_events())
+        }
+
+        multi_groups: Mapping[int, Tuple[GroupEvent, GroupState]] = {
+            gs.get("id"): (group_events.get(gs.get("id")), gs)
+            for gs in group_states
+            if gs.get("id") is not None
+        }
```

Comment:
does this need to be in a dict if we're going to ignore the key for `group_jobs`?